core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -432,8 +432,10 @@ private[spark] class ExecutorAllocationManager(
     if (testing || executorsRemoved.nonEmpty) {
       executorsRemoved.foreach { removedExecutorId =>
         newExecutorTotal -= 1
+        val hasCachedBlocks = SparkEnv.get.blockManager.master.hasCachedBlocks(removedExecutorId)
```

Comment:
This is a pretty expensive call just so you can print a more accurate log message... at the very least it should be inside the `logInfo` call:

```
logInfo {
  expensiveStuff()
  "This is the log message referencing the expensive stuff."
}
```

But I'd prefer a solution that doesn't make this call at all, since INFO is more often than not enabled.